### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by AShozi**
Linting is giving me errors in all my branches so its being temporarily removed as discussion with mentors until issue can be resolved

### CODE REVIEW

File: phantomgames/Search/SearchGameViewModel.swift [hunk 1]
```diff
@@ -21,8 +21,8 @@ class SearchGameViewModel {
     private(set) var allGameList: [Game] = []
     private(set) var filteredGames: [Game] = []
     
-    init(repository: SearchGameRepositoryType, delegate: ViewModelDelegate) { self.repository =
-        repository;self.delegate = delegate }
+    init(repository: SearchGameRepositoryType, delegate: ViewModelDelegate) { 
+        self.repository = repository;self.delegate = delegate }
```
Comment (simekanidev) [inline phantomgames/Search/SearchGameViewModel.swift]:
```suggestion
        self.repository = repository
        self.delegate = delegate
       }
```

File: phantomgamesTests/SearchGameViewModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+@testable import phantomgames
+
+final class SearchGameViewModelTests: XCTestCase {
+    
+    var viewModel: SearchGameViewModel!
+    var mockRepository: MockSearchGameRepository!
+    var mockDelegate: MockViewModelDelegate!
+    
+    override func setUp() {
+        mockRepository = MockSearchGameRepository()
```
Comment (simekanidev) [inline phantomgamesTests/SearchGameViewModelTests.swift:19]: super.setUp()

File: phantomgames/Search/SearchGameViewModel.swift (hunk 1 above)
Comment (kieran-woodrow) [reply phantomgames/Search/SearchGameViewModel.swift]: swiftlint should be picking up these as warnings. Please don't ignore them

File: phantomgamesTests/SearchGameViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    func testGamelistCount () {
+        viewModel.fetchSearchResults()
+        let result = viewModel.gameListCount
+        XCTAssertEqual(result, 2)
+    }
+    
+    func testGameAtIndex () {
+        viewModel.fetchSearchResults()
+        
+        let expectedResult = Game(id: 1, title: "game1", thumbnail: "Thumbnail1", shortDescription: "shortDes1", gameURL: "URL1", genre: "genre1", platform: "platform1", publisher: "publisher1", developer: "dev1", releaseDate: "Date1", freetogameProfileURL: "Profile1").title
```
Comment (simekanidev) [inline phantomgamesTests/SearchGameViewModelTests.swift]: Expected result can be removed since it's not used

**Review by kieran-woodrow — APPROVED** [review]

**Review by kiasha10 — APPROVED** [review]

**Pull request merged**